botocore/configloader.py

```diff
@@ -142,12 +143,13 @@ def raw_config_parse(config_filename, parse_subsections=True):
         path = os.path.expandvars(path)
         path = os.path.expanduser(path)
         if not os.path.isfile(path):
-            raise botocore.exceptions.ConfigNotFound(path=path)
-        cp = configparser.RawConfigParser()
+            raise botocore.exceptions.ConfigNotFound(path=_unicode_path(path))
+        cp = six.moves.configparser.RawConfigParser()
         try:
-            cp.read(path)
-        except configparser.Error:
-            raise botocore.exceptions.ConfigParseError(path=path)
+            cp.read([path])
```

Comment:
This was needed to be able to write tests that pass in bytes. `read` takes in a list of file paths, but if you provide a single string it will wrap that in a list. But python 3 treats strings and bytes differently, and in the code specifically for `read` it doesn't take bytes into account (which is probably a bug). But since a `bytes` object is iterable, it just silently does the wrong thing.

This is needed so that tests can be written that pass in raw bytes and have them function on both python 2 and python 3.